src/sysen2/mldev.108

```diff
@@ -166,11 +207,39 @@ notits:	;; nothing came of it, fall through
 			.SUSET [.RUNAME,,JBCFN2]
 			JRST (A) ]
 		MOVEI A,'ML		;USE THE MATHLAB LINE PRINTER
-		JRST GOTHST ]
+		JRST GOTHSX ]
 ;HOST NOT RECOGNIZED
 NSD:	MOVSI A,%ENSDV
 	JRST NOGO2
 
+hhhddd:                 ; get device from last 3 characters
+	ldb c,[002200,,a]   ; get device
+	hrlz c,c            ; left-jusify
+	movem c,jbcdev      ; save remote device
+	ldb b,[222200,,a]   ; get hostname
+	hrlz a,b            ; left-justify
+	jrst gothst         ; process host
+
+doshoe:                 ; device is SHOE
+	lsh c,12.	        ; left justfify device
+	movem c,jbcdev		; set remote device
+	ldb b,[301400,,a]	; get 2-character hostname
+	lsh b,30		    ; left justify hostname
+
+chkhst:                 ; b has host
+	pushj p,isitsp      ; is it a known ITS machine
+	  jrst .+1          ; ???
```

Comment:
No need to call `isitsp` if the result isn't used.

Reply:
The old code did in a bunch of places. Didn’t really know why. Left it.